pypdf/_page.py

```diff
@@ -1914,9 +2004,35 @@ def extract_text(
                 The font-dictionary may be None in case of unknown fonts.
                 If not None it may e.g. contain key "/BaseFont" with value "/Arial,Bold".
 
+        KwArgs:
+            extraction_mode (Literal["plain", "layout"]): "plain" for legacy functionality
+                "layout" for experimental layout mode functionality.
+                NOTE: orientations, space_width, and visitor_* parameters are NOT respected
+                in "layout" mode.
+            layout_mode_space_vertically: include blank lines inferred from y distance + font
+                height. Defaults to True.
+            layout_mode_scale_weight: multiplier for string length when calculating weighted
+                average character width. Defaults to 1.25.
+            layout_mode_debug_path (Path | None): if supplied, must target a directory.
+                creates the following files with debug information for layout mode
+                functions if supplied:
+                  - fonts.json: output of self._layout_mode_fonts
+                  - tjs.json: individual text render ops with corresponding xform matrices
+                  - bts.json: text render ops left justified and grouped by BT/ET operators
+                  - bt_groups.json: BT/ET operations grouped by rendered y-coord (aka lines)
+
         Returns:
             The extracted text
         """
+        extraction_mode: Literal["plain", "layout"] = kwargs.get("extraction_mode", "plain")
```

Comment:
Why didn't you add a new parameter `extraction_mode: Literal["plain", "layout"] = "plain"` to the function signature?

Reply:
No reason really. I had it the other way initially. Switched back to an explicit "extraction_mode" parameter but left the other layout mode settings as kwargs for now.